camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java

```diff
@@ -275,7 +278,16 @@ private void runCollectorForTopicDir(FileSystem fs, String topic, Path topicSour
     List<Properties> jobPropsList = planner.createSweeperJobProps(topic, topicSourceDir, topicDestDir, fs);
 
     for (Properties jobProps : jobPropsList){
-      tasksToComplete.add(runCollector(jobProps, topic));
+      Future<?> task = runCollector(jobProps, topic);
+      tasksToComplete.add(task);
+
+      try {
+        task.get();
```

Comment:
hmm shouldn't this blocking call be outside of the loop?
like https://github.com/vinted/warehouse-jobs/pull/925#discussion_r64574115